### ISSUE TITLE
Grid needs documentation about a11y limitation of using flex-order

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3412]
Flexbox ordering is not acknowledged by screen readers, so we should make a note in our documentation that DOM order is the preferred way to ensure screen readers get content in the right order. There may be uses for the column ordering that wouldn't break accessibility, but we need to properly document this for consumers.

**Comment by chlane** [comment]
We should remove the Column Ordering section https://clarity.design/documentation/grid. Screen reader users should be able to read the content in the intended order.  If the visual design has content in a certain order, the content's DOM order should match. The flexbox order property, if used as intended creates a different reading order for sighted users than for screen reader users which violates WCAG 2.1  1.3.2 Meaningful Sequence(https://www.w3.org/TR/WCAG21/#meaningful-sequence)
